### PR TITLE
Consolidate script editors, add analysis.R

### LINKED ISSUE #58: Integrate `webr` and Pyodide to allow analysis of the samples
Similar to #57, but rather than before the sampling run, this would be done afterward. 

Ideally this would look like a single script file and an output area that shows both stdout/stderr **and plots**. See https://www.cararthompson.com/posts/2024-02-01-experimenting-with-webr/ for some prior art on this.

A pre-populated magic variable called `samples` would make sense here. In terms of libraries to support, I think a nice target use-case is being able to use `posterior` + `bayesplot` in R, and `arviz` in Python.

### LINKED ISSUE #106: Attempt to unify editor components
In the current main branch, we have separate editor components for stan code and data code, and (while I'm not caught up on the current PRs) I believe that will also be the case for the R and Python editors.

It'd be great if we could avoid repeating major parts of these components when their contents are largely similar. I haven't investigated all the details of how to achieve this, but we should look at it after feature completion for 1.0.

### PR DESCRIPTION
**Pull request by WardBrian**
I decided to open this as a replacement to PR #155 rather than just pushing to that branch because I did a bunch of code moves.

This PR:

- Adds two new components, `ScriptEditor` and its big brother, `PlottingScriptEditor`. These are the basis of the data and analysis tabs.
- Refactored the existing tabs to share as much code as is reasonable, and use those components
- Reorganize the folder structure of this code (everything that touches webr/pyodide now entirely lives under `app/Scripting`) and move any large R/Python code chunks out of strings and into their own dedicate files.
- Add the analysis.R window, with the loading code for the draws based on posterior.

Only this final bullet point changes any of the functionality, the rest is just tech debt cleanup. 

Closes #106 
Closes #58

### CODE REVIEW

**Comment by magland** [comment]
Very Cool!

Before I start reading the code, I wanted to test to see if everything is working. One thing I noticed (I think it was introduced in this PR). When I run the python data generation script it always needs to reload pyodide. Is the worker getting terminated?

data.R does not have this problem.

analysis.py and analysis.R also do not have this problem.

**Comment by WardBrian** [comment]
@magland I observe the behavior you are describing on the current `main` at https://stan-playground.flatironinstitute.org/, so I don't think it originates here. I have a hunch about where it would come from, let me see if I'm write.

Otherwise, another point in favor of #142, possibly

**Comment by WardBrian** [comment]
@magland I believe it is fixed. Basically, the onData part of the callback was changing after every run, since it depended on the data, so it re-loaded the usePyodide hook.

**Review by magland — APPROVED** [review]
Just a few minor comments. We should merge this soon because there are so many changes and renames.

**Comment by WardBrian** [comment]
@magland done, and I agree I'd prefer to merge this before too much else just due to conflicts. I'd still appreciate if @jsoules could take a look, since some of the consolidation was a request of his

File: gui/src/app/Scripting/Analysis/useAnalysisState.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  const consoleRef = useRef<HTMLDivElement | null>(null);
+  const imagesRef = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    if (imagesRef.current) {
+      imagesRef.current.innerHTML = "";
+    }
+    if (consoleRef.current) {
+      consoleRef.current.innerHTML = "";
+    }
```
Comment (jsoules) [inline gui/src/app/Scripting/Analysis/useAnalysisState.ts:27]: This shows up here and in `AnalysisPyWindow.tsx`, maybe pop it in a function?

**Review by jsoules — APPROVED** [review]
I think this looks fine. There's a couple minor things that might be slightly cleaner, but I'd also be happy to just merge as is.

File: gui/src/app/Scripting/Analysis/useAnalysisState.ts (hunk 1 above)
Comment (WardBrian) [reply gui/src/app/Scripting/Analysis/useAnalysisState.ts]: Done

File: gui/src/app/Scripting/Analysis/AnalysisPyWindow.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+        const img = document.createElement("img");
+        img.style.width = "100%";
+        img.src = imageUrl;
+
+        const divElement = document.createElement("div");
+        divElement.appendChild(img);
+        imagesRef.current?.appendChild(divElement);
+      },
+      onStatus,
```
[the remark this answers is not in the document]
Comment (WardBrian) [reply gui/src/app/Scripting/Analysis/AnalysisPyWindow.tsx]: Didn't love how this looked, but I did pull the onImage out into a function

[the remark this answers is not in the document]
Comment (jsoules) [reply gui/src/app/Scripting/Analysis/AnalysisPyWindow.tsx]: No? If you don't like the look of it, feel free to not do it :)

**Pull request merged**